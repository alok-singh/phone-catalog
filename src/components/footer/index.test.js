import React from "react";
import Footer from "./index";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

const renderApp = (props) => {
  return render(
    <BrowserRouter>
      <Footer {...props} />
    </BrowserRouter>
  );
};

beforeEach(() => {});

afterEach(() => {
  cleanup();
});

test("Footer renders correctly", () => {
  const props = {
    list: [{link: "link", title: "title"}],
    copyright: 'copyright'
  };
  
  const { container } = renderApp(props);
  expect(container.firstChild).toMatchSnapshot();
});
