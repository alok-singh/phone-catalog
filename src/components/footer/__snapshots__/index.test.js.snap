// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  class="footer-wrapper"
>
  <div
    class="footer-liks"
  >
    <a
      class="footer-item"
      href="/link"
    >
      <div
        class="footer-text"
      >
        title
      </div>
    </a>
  </div>
  <div
    class="copyright-text"
  >
    copyright
  </div>
</div>
`;
